src/components/SearchInterface.tsx

```diff
@@ -22,151 +24,183 @@ interface Source {
-                      {sortBy}
-                      <ChevronDown className="ml-2 h-4 w-4" />
-                    </Button>
-                  </DropdownMenuTrigger>
-                  <DropdownMenuContent align="end">
-                    <DropdownMenuItem onSelect={() => setSortBy('Most Recent')}>Most Recent</DropdownMenuItem>
-                    <DropdownMenuItem onSelect={() => setSortBy('Oldest')}>Oldest</DropdownMenuItem>
-                    <DropdownMenuItem onSelect={() => setSortBy('Most Popular')}>Most Popular</DropdownMenuItem>
-                  </DropdownMenuContent>
-                </DropdownMenu>
-              </div>
-              <div className="overflow-y-auto flex-grow">
-                {snippets.map((snippet) => (
-                  <div key={snippet.id} className="mb-4 p-4 border rounded-lg cursor-pointer" onClick={() => handleSnippetClick(snippet.id)}>
-                    <div className="flex justify-between items-start mb-2">
-                      <div>
-                        <h3 className="text-lg font-semibold">ID {snippet.id} {snippet.radio_code}</h3>
-                        <p className="text-sm text-gray-500">{snippet.name}</p>
-                      </div>
-                      <div className="flex space-x-2">
-                        <Button variant="ghost" size="icon">
-                          <Share2 className="h-4 w-4" />
-                        </Button>
-                        <Button variant="ghost" size="icon">
-                          <Star className="h-4 w-4" />
-                        </Button>
-                      </div>
-                    </div>
-                    <p className="mb-4">{snippet.type} - {snippet.channel}</p>
-                    <div className="flex space-x-2 mb-4">
-                      <Button variant="outline" size="sm" className="rounded-full bg-blue-100 text-blue-600">
-                        {snippet.state} <ArrowUp className="ml-1 h-3 w-3" /> {snippet.human_upvotes}
-                      </Button>
-                    </div>
-                    <LiveblocksComments
-                      snippetId={snippet.id.toString()}
-                      showFullComments={false}
-                    />
-                  </div>
-                ))}
-              </div>
-            </div>
+          <h3 className="font-medium mt-6 mb-2">State</h3>
+          <MultiSelectDropdown
+            selectedItems={states}
+            items={STATES}
+            onItemToggle={(state: string) => handleMultiSelect(setStates, STATES, state)}
+            placeholder="Select states"
+            allItemsLabel={STATES[0]}
+          />
+
+          <h3 className="font-medium mt-6 mb-2">Source</h3>
+          <MultiSelectDropdown
+            selectedItems={sources}
+            items={RADIO_STATIONS}
+            onItemToggle={(source: string) => handleMultiSelect(setSources, RADIO_STATIONS, source)}
+            placeholder="Select languages"
```

Comment:
--- $${\color{green}[AI \space Generated \space Comment]}$$ ---
The placeholder text for the 'Source' dropdown is incorrect. It should be changed from "Select languages" to "Select sources":

```typescript
placeholder="Select sources"
allItemsLabel={RADIO_STATIONS[0]}
```